### ISSUE TITLE
Add more templates?

### ISSUE THREAD

**Issue by zkamvar** [ismayc/thesisdown#4]
Hi Chester,

As you saw in #2, I have an interest in utilizing this for the Oregon State University thesis template.  My work is currently on my [osu-thesis][0] branch. I'm wondering if you would be up for expanding this package in a similar vein to [rticles][1]. Plus, if we get people from PSU and UO to contribute, we could have an oregonthesis package on our hands 😸 . What do you think?

[0]: https://github.com/zkamvar/thesisdown/tree/osu-thesis
[1]: https://github.com/rstudio/rticles

**Comment by ismayc** [comment]
Hi @zkamvar.  I have thought about this quite a bit.  The unfortunate thing comes (as I'm sure you've noticed by now) in setting up the cls and template.tex files appropriately to interact with the YAML.  I do like the idea of generalizing as you did in your YAML for `degree`, `depttype`, etc., but without having all of the LaTeX templates available from a bunch of different schools/degrees it's gonna be a little tricky to generalize and potentially even more painful to dig through the LaTeX templates of other schools.  I'd also prefer to not have a huge list of templates to choose from initially whenever someone installs `thesisdown`.

`rticles` is a good idea, but if we have just a subtle little tweak between two different template files and a listing of all the different types of degrees and names of schools, it may be pretty difficult for folks to find the exact template they need.

At this time, I'd rather have folks look over the **template.tex** and **reedthesis.cls** files and customize for their own school.  A repository of thesis template.tex and cls files is an enticing idea though... Maybe we should think and talk more about it.

**Comment by zkamvar** [comment]
I completely agree on steering thousands of miles clear of attempting to generalize LaTeX templates across universities. I was thinking more along the lines of the latter idea:

>  A repository of thesis template.tex and cls files is an enticing idea though

If that ends up with a package that appears to have too many thesis skeletons 💀 , perhaps a README on how to create ones own template provided their university already has a LaTeX-based one? The [documentation on creating LaTeX templates][0] is sorely lacking. 

Again, I can't say how appreciative I am that you had this idea and carried out a fantastic working example. I've tried all three ways of authoring manuscripts with my advisor (Word, LaTeX, and RMarkdown), and by far, this method has been the easiest.

[0]: http://rmarkdown.rstudio.com/pdf_document_format.html#custom_templates

**Comment by ismayc** [comment]
Excellent idea!  I am going to have a busy few months ahead, but would love to work on some documentation for LaTeX templates and write some R functions to assist with this.  It took me a few weeks to figure out how to get them to work when I wrote the reedtemplates package originally.  You are right that it is lacking big time.

**Comment by ismayc** [comment]
I am glad this is a useful resource for you.  I highly recommend producing a gitbook version of your thesis as well and publishing it to bookdown.org or github.io.

**Comment by mikey-harper** [comment]
Hi, great work with this project! It has been a great starting block for me designing my own template for my thesis.

I have designed my own template based on the University of Southampton Thesis: [thesis.pdf](https://github.com/ismayc/thesisdown/files/1066779/thesis.pdf). I was wondering what you preferred me doing:

1) adding a second template to your existing package. I have designed the template to be as generic as possible (all the university details are inputted within the YAML).
2) Keep it as a separate project on my GitHub like this: https://github.com/mikey-harper/thesisdown . Obviously would need to alter it to give credit to you!

Merging the two projects into one might help to keep them up to date more. For example, I made a few alterations:
- edited the pandoc code to remove depricated functions.
- altered the YAML to include more options (font type etc.)

Happy to help further develop this where possible. I found it quite confusing to understand at first but feel I have got my head around things a bit more now!

**Comment by ismayc** [comment]
Hi @mikey-harper.  Thanks so much for reaching out.  I'm hoping to figure out a good solution to this sometime this summer.  For the time being, I can see your fork.

If you could document what you found particularly confusing (maybe in a shared Google Doc?), @nicksolomon and I can work with you on improving the package for others to use down the road.

**Comment by mikey-harper** [comment]
I'd be happy to fully document what I found hard and try and help write some guidance on. Super busy at the moment so you'll have to do with my comments in note form for now, but hopefully expand on them once I get more time:

**Overview of my level of experience:** relatively new to LaTeX, having used it for the past year. This was my first proper time delving into building any form of template. I basically merged your project with a LaTeX template I found for my uni. 

My main issues I faced:
- Not fully understanding which bits of code in the thesisdown template and cls file were needed to make all the formatting work vs. code which was bespoke to your layout.  You could section the template slightly differently so that users know which code should not be deleted (i.e. most the \newcommand commands)
- Not fully appreciating the power of pandoc in setting variables. The $if$ statements make the LaTeX template look quite intimidating at the first  so a short explanation about this would probably be useful
- Troubleshooting through RStudio is rubbish as you get very cryptic error codes (i,e, pandoc failed with error 1). While people are building their templates, I would recommend that they use something like Texmaker which gives detailed error messages. 

Hope these make sense. Let me know if you need me to clarify, and happy to potentially help out on writing any user guide?

**Comment by ismayc** [comment]
Thanks, @mikey-harper. Ideally, I'd like to write R functions using regular expressions to be able to go through and find the needed spots in a user's college/university template without having the `thesisdown` user have to interact with LaTeX at all.  That seems like a pretty challenging task but it would go a long way in helping more people have reproducible theses using R Markdown.

For the time being, a user guide detailing what steps you needed to make would really help in generalizing this process for other theses.  As thorough as possible with screenshots as needed would be great.  That would really get the ball rolling.  Thanks again!

**Comment by achamess** [comment]
Thanks for getting this going. I’ll look in some of the other offshoot projects.

Sent from my iPhone

On May 12, 2018, at 5:23 PM, Chester Ismay <notifications@github.com<mailto:notifications@github.com>> wrote:


I also recommend checking out https://github.com/benmarwick/huskydown<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_benmarwick_huskydown&d=DwMFaQ&c=imBPVzF25OnBgGmVOlcsiEgHoG1i6YHLR0Sj_gZ4adc&r=j_eoU9TVYa9LTnnoTWWpgQ&m=llZpKJsK2PTaCC6BeX8WfmQqiO0PTJFRYfBk53p_rtM&s=CCC7BmakmUHZaAgNmFSFzlcW1db3_iiwAWQgb4Vjlwc&e=>
